server/api.go

```diff
@@ -582,17 +620,65 @@ func (p *Plugin) createIssueComment(w http.ResponseWriter, r *http.Request) {
 		githubClient = p.githubConnect(*info.Token)
 	}
 
+	api := p.API
+	post, appErr := api.GetPost(req.PostId)
+	if appErr != nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + req.PostId, StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + req.PostId + ": not found", StatusCode: http.StatusNotFound})
+		return
+	}
+
+	commentUser, appErr := api.GetUser(post.UserId)
+	if appErr != nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post.UserID " + post.UserId + ": not found", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	currentUser, appErr := api.GetUser(userID)
+	if appErr != nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load current user", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	siteUrl := api.GetConfig().ServiceSettings.SiteURL
+
+	permalink := getPermaLink(*siteUrl, req.PostId, req.CurrentTeam)
+
+	permalinkMessage := fmt.Sprintf("*@%s attached a* [message](%s) *from @%s*\n", currentUser.Username, permalink, commentUser.Username)
+
+	req.Comment = permalinkMessage + req.Comment
 	comment := &github.IssueComment{
 		Body: &req.Comment,
 	}
 
-	result, _, err := githubClient.Issues.CreateComment(ctx, req.Owner, req.Repo, req.Number, comment)
-	if err != nil {
-		mlog.Error(err.Error())
-	}
+	if result, rawResponse, err := githubClient.Issues.CreateComment(ctx, req.Owner, req.Repo, req.Number, comment); err != nil {
```

Comment:
Generally we like to not use the else block like this. See: https://golang.org/doc/effective_go.html#if

Reply:
Done without else block!